_inc/client/components/data/query-products/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isDevMode } from 'state/connection';
+import { fetchProducts, isFetchingProducts } from 'state/site';
+
+class QueryProducts extends Component {
+	static propTypes = {
+		isFetchingProducts: PropTypes.bool,
+		isDevMode: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		isFetchingSiteBenefitsData: false,
+	};
+
+	componentDidMount() {
+		if ( ! this.props.isDevMode && ! this.props.isFetchingSiteBenefits ) {
```

Comment:
Why do we care about dev mode here? It doesn't make much sense to me. If there's no obvious reason, let's remove it to keep this one simple.

Reply:
Since so many other components check for it I assumed it was important, but I've removed it now.